### PR TITLE
Doxygen document generation

### PR DESCRIPTION
**Pull request by Ashu7950**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #[Issue]

**Description**

Precise description of the changes in this pull request

**Alternative(s) considered**

Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**

Type Choose one: (Bug fix | Feature | Documentation | Testing | Other)

**Screenshots (if applicable)**

**Checklist**

- [x] I have read the [Contribution Guidelines](https://github.com/Apra-Labs/ApraPipes/wiki/Contribution-Guidelines)
- [x] I have written Unit Tests
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach

### CODE REVIEW

**Review by yashrajsapra — COMMENTED** [review]
@joiskash @mraduldubey 

Should we build documentation along with ApraPipes, or should it be separate?

File: build_scripts/build_dependencies_jetson_cuda.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # List of required dependencies
-dependencies=("git-lfs" "libncurses5-dev" "ninja-build" "nasm" "curl" "libudev-dev" "libssl-dev")
+dependencies=("git-lfs" "libncurses5-dev" "ninja-build" "nasm" "curl" "libudev-dev" "libssl-dev" "doxygen" "graphviz")
```
Comment (yashrajsapra) [inline build_scripts/build_dependencies_jetson_cuda.sh:4]: @kushaljain-apra 
Do we need graphviz for doxygen?

File: build_scripts/build_dependencies_windows_cuda.ps1 [hunk 2]
```diff
@@ -5,7 +5,29 @@ iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/in
 
 # Enable feature and install dependencies
 choco feature enable -n allowEmptyChecksums
-choco install 7zip git python3 cmake pkgconfiglite -y
+choco install 7zip git python3 cmake pkgconfiglite doxygen.install graphviz -y
+
+# Specify the new path to be added to the PATH environment variable
+$doxygenPath = "C:\Program Files\doxygen\bin"
```
Comment (yashrajsapra) [inline build_scripts/build_dependencies_windows_cuda.ps1:11]: Please don't use hardcoded path

File: build_scripts/build_dependencies_windows_cuda.ps1 [hunk 3]
```diff
@@ -33,4 +55,9 @@ if (-not (Test-Path "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\$cudaVer
     Copy-Item -Path "C:\Users\$userName\Downloads\cudnn-*\bin\*.dll" -Destination "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\$cudaVersion\bin\" -Recurse
 }
 
-Write-Host "Dependencies verified and installed successfully."
\ No newline at end of file
+Write-Host "Dependencies verified and installed successfully."
+
+# Build Documentation
```
Comment (yashrajsapra) [inline build_scripts/build_dependencies_windows_cuda.ps1:60]: We Should not build documentation while building dependencies

File: build_scripts/build_dependencies_windows_no_cuda.ps1 [hunk 4]
```diff
@@ -14,4 +36,9 @@ pip3 install meson
 # Install cmake
 pip3 install cmake --upgrade
 
-Write-Host "Dependencies verified and installed successfully."
\ No newline at end of file
+Write-Host "Dependencies verified and installed successfully."
+
+# Build Documentation
```
Comment (yashrajsapra) [inline build_scripts/build_dependencies_windows_no_cuda.ps1:41]: We Should not build documentation while building dependencies

File: build_scripts/build_dependencies_windows_no_cuda.ps1 [hunk 5]
```diff
@@ -5,7 +5,29 @@ iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/in
 
 # Enable feature and install dependencies
 choco feature enable -n allowEmptyChecksums
-choco install 7zip git python3 cmake pkgconfiglite -y
+choco install 7zip git python3 cmake pkgconfiglite doxygen.install graphviz -y
+
+# Specify the new path to be added to the PATH environment variable
+$doxygenPath = "C:\Program Files\doxygen\bin"
```
Comment (yashrajsapra) [inline build_scripts/build_dependencies_windows_no_cuda.ps1:11]: Please don't use a hardcoded path

File: build_scripts/build_dependencies_jetson_cuda.sh (hunk 1 above)
Comment (kushaljain-apra) [reply build_scripts/build_dependencies_jetson_cuda.sh:4]: Yes, the graphs that are being generated for each class uses graphviz.

Comment (yashrajsapra) [reply build_scripts/build_dependencies_jetson_cuda.sh:4]: Okay

File: build_scripts/build_dependencies_windows_cuda.ps1 (hunk 2 above)
Comment (kushaljain-apra) [reply build_scripts/build_dependencies_windows_cuda.ps1]: Resolved.

File: build_scripts/build_dependencies_windows_cuda.ps1 (hunk 3 above)
Comment (kushaljain-apra) [reply build_scripts/build_dependencies_windows_cuda.ps1]: Removed documentation build from scripts.

File: build_scripts/build_dependencies_windows_no_cuda.ps1 (hunk 5 above)
Comment (kushaljain-apra) [reply build_scripts/build_dependencies_windows_no_cuda.ps1]: Resolved.

File: build_scripts/build_dependencies_windows_no_cuda.ps1 (hunk 4 above)
Comment (kushaljain-apra) [reply build_scripts/build_dependencies_windows_no_cuda.ps1]: Removed documentation build from scripts.

**Comment by mraduldubey** [comment]
> @joiskash @mraduldubey
> 
> Should we build documentation along with ApraPipes, or should it be separate? Like separate script for building document

Both. 
- Launch the doc generation script from the build script 
- Add instructions on readme on how to build only docs separately

**Review by mraduldubey — CHANGES_REQUESTED** [review]
Is it possible to show the latest commit hash in the footer of the html pages/doxy output ?

**Review by joiskash — APPROVED** [review]

**Comment by yashrajsapra** [comment]
> Is it possible to show the latest commit hash in the footer of the html pages/doxy output ?

Part of Commit [1a61333](https://github.com/Apra-Labs/ApraPipes/pull/337/commits/1a61333ee51acbb9f944e132a2c00cee1a7d27d6)

**Comment by yashrajsapra** [comment]
> Is it possible to show the latest commit hash in the footer of the html pages/doxy output ?

Part of Commit [1a61333](https://github.com/Apra-Labs/ApraPipes/pull/337/commits/1a61333ee51acbb9f944e132a2c00cee1a7d27d6)

**Review by mraduldubey — APPROVED** [review]

**Pull request merged**